lib/queue.rb

```diff
@@ -1,31 +1,71 @@
 class Queue
-
+  attr_reader :front, :back, :store
   def initialize
-    # @store = ...
-    raise NotImplementedError, "Not yet implemented"
+    @size = 20
+    @store = Array.new(@size)
+    @front = -1
+    @back = -1
   end
 
   def enqueue(element)
-    raise NotImplementedError, "Not yet implemented"
+    if empty?
+      @front = 0
+      @back = 0
+    elsif @front == @back
+      raise StandardError.new "Queue is out of room! Needs to be resized."
+    end
+
+    @store[@back] = element
+    @back = (@back + 1) % @store.length
   end
 
   def dequeue
-    raise NotImplementedError, "Not yet implemented"
+    return nil if empty?
+    
+    removed = @store[@front]
+    @store[@front] = nil
+
+    if @back - 1 == @front
+      @front = -1
+      @back = -1
+    else
+      @front = (@front + 1) % @store.length
+    end
+    return removed
   end
 
   def front
-    raise NotImplementedError, "Not yet implemented"
+    return nil if empty?
+    return @front
   end
 
   def size
-    raise NotImplementedError, "Not yet implemented"
+    return 0 if empty?
+    return @front - @back if @front > @back
+    return @back + (@store.length - @front)
```

Comment:
I think you have this reversed here.  

```suggestion
    return @back - @front if @front < @back
    return @back + (@store.length - @front)
```